EdgeMultiplay/Scripts/EdgeMultiplayObserverLocal.cs

```diff
@@ -0,0 +1,207 @@
+        private NetworkedPlayer networkPlayer;
+        private Vector3 targetPosInterpolated;
+        private Vector3 targetRotInterpolated;
+
+        #endregion
+
+        #region MonoBehaviour Callbacks
+
+#if UNITY_EDITOR
+
+        //Reset is a Monobehaviour callback and is called once a Component is added to a GameObject
+        [ExecuteAlways]
+        void Reset()
+        {
+            eventId = Guid.NewGuid().ToString("N") + gameObject.GetInstanceID().ToString();
+        }
+
+#endif
+        private void Start()
+        {
+            if (!attachedToPlayer)
+            {
+                LocalPlayeIsMaster = EdgeManager.localPlayerIsMaster;
+            }
+            EdgeManager.observers.Add(this);
+            networkPlayer = GetComponent<NetworkedPlayer>();
+            positionFromServer = transform.localPosition;
+            rotationFromServer = transform.localRotation.eulerAngles;
+        }
+
+        private void Update() 
+        {
+            if (EdgeManager.gameStarted)
+            {
+                if ((latestPosition != transform.localPosition && syncOption == SyncOptions.SyncPosition)
+                || (latestRotation != transform.localRotation.eulerAngles && syncOption == SyncOptions.SyncRotation)
+                || ((latestRotation != transform.localRotation.eulerAngles || latestPosition != transform.localPosition) && syncOption == SyncOptions.SyncPositionAndRotation))
+                {
+                    if (attachedToPlayer)
+                    {
+                        if (networkPlayer && networkPlayer.isLocalPlayer)
+                        {
+                            SendDataToServer(syncOption, true, networkPlayer.playerId);
+                        }
+                    }
+                    else if (!attachedToPlayer && LocalPlayeIsMaster)
+                    {
+                        SendDataToServer(syncOption, false, eventId);
+                    }
+                }
+            }
+        }
+
+        private void LateUpdate()
+        {
+            if (attachedToPlayer)
+            {
+                if (networkPlayer && !networkPlayer.isLocalPlayer)
+                {
+                    ReflectServerData(syncOption);
+                }
+            }
+            else if (!attachedToPlayer && !LocalPlayeIsMaster)
+            {
+                ReflectServerData(syncOption);
+            }
+        }
+
+        private void OnDestroy()
+        {
+            eventId = null;
+        }
+
+        #endregion
+
+        #region EdgeMultiplayObserver Private Functions
+      
+        void ReflectServerData(SyncOptions syncOption)
+        {
+            switch (syncOption)
+            {
+                case SyncOptions.SyncPosition:
+                    if (InterpolatePosition)
+                    {
+                        targetPosInterpolated = Vector3.Lerp(transform.localPosition, positionFromServer, Time.deltaTime * InterpolationFactor);
```

Comment:
Lerp is great. Where is InterpolationFactor?

Reply:
EdgeMultiplayObserverLocal inherits from EdgeMultiplayObserver, InterpolationFactor is defined in EdgeMultiplayObserver and sat in the inspector as a float range